SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -1684,35 +1684,78 @@ USpatialActorChannel* ActorSystem::TryRestoreActorChannelForStablyNamedActor(AAc
 {
-	for (FObjectRepNotifies& ObjectRepNotifies : RepNotifiesToSend)
+	// Call rep notifies on actors before subobjects
+	TArray<FObjectRepNotifies> Actors;
+	TArray<FObjectRepNotifies> Subobjects;
+	Actors.Reserve(RepNotifiesToSend.Num());
+	Subobjects.Reserve(RepNotifiesToSend.Num());
+
+	for (auto& ObjectToObjRepNotifies : RepNotifiesToSend)
 	{
-		UObject* Object = ObjectRepNotifies.Object.Get();
-		if (!IsValid(Object))
+		FObjectRepNotifies& ObjectRepNotifies = ObjectToObjRepNotifies.Value;
+		if (ObjectRepNotifies.RepNotifies.Num() == 0)
 		{
 			continue;
 		}
-		const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Object);
-		if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+
+		UObject* Object = ObjectToObjRepNotifies.Key.Get();
+		ObjectRepNotifies.Object = Object;
+
+		if (Cast<AActor>(Object))
 		{
-			// TODO UNR-5785 - Once fixed, change log level to warning
-			UE_LOG(LogActorSystem, Log, TEXT("Failed to invoke rep notifies for an object as its entity id was invalid. Object: %s"),
-				   *Object->GetName());
-			continue;
+			Actors.Add(MoveTemp(ObjectRepNotifies));
 		}
-		USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(EntityId);
-		if (!IsValid(Channel))
+		else
 		{
-			UE_LOG(LogActorSystem, Warning,
-				   TEXT("Failed to invoke rep notifies for an object as its channel was invalid. Object: %s, Entity: %lld"),
-				   *Object->GetName(), EntityId);
-			continue;
+			Subobjects.Add(MoveTemp(ObjectRepNotifies));
 		}
-		RemoveRepNotifiesWithUnresolvedObjs(*Object, *Channel, ObjectRepNotifies.RepNotifies);
-		Channel->PostReceiveSpatialUpdate(Object, ObjectRepNotifies.RepNotifies, ObjectRepNotifies.PropertySpanIds);
 	}
+
+	for (FObjectRepNotifies& ActorRepNotifies : Actors)
+	{
+		TryInvokeRepNotifiesForObject(ActorRepNotifies);
+	}
+	for (FObjectRepNotifies& ObjectRepNotifies : Subobjects)
+	{
+		TryInvokeRepNotifiesForObject(ObjectRepNotifies);
+	}
+
 	RepNotifiesToSend.Empty();
 }
 
+void ActorSystem::TryInvokeRepNotifiesForObject(FObjectRepNotifies& ObjectRepNotifies) const
+{
+	UObject* Object = ObjectRepNotifies.Object;
+	if (!IsValid(ObjectRepNotifies.Object))
+	{
+		return;
+	}
+	const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Object);
+	if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		// TODO UNR-5785 - Once fixed, change log level to warning
+		UE_LOG(LogActorSystem, Log, TEXT("Failed to invoke rep notifies for an object as its entity id was invalid. Object: %s"),
+			   *Object->GetName());
+		return;
+	}
+	USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(EntityId);
+	if (!IsValid(Channel))
+	{
+		UE_LOG(LogActorSystem, Warning,
+			   TEXT("Failed to invoke rep notifies for an object as its channel was invalid. Object: %s, Entity: %lld"), *Object->GetName(),
+			   EntityId);
+		return;
+	}
+
+	Algo::Sort(ObjectRepNotifies.RepNotifies, [](GDK_PROPERTY(Property) * A, GDK_PROPERTY(Property) * B) -> bool {
```

Comment:
Here you are using Algo::Sort, which is providing pointers from the array. And below you are assuming these pointers are always valid and not verifying they are null before accessing them. If they can be null, then you will want to verify this, else you'll get null pointer accesses. If they cannot be null, then you might find TArray::Sort is a better choice since the predicate will be passed references rather than pointers.

Reply:
Yup, they can't be null, changed to tarr::sort